### ISSUE TITLE
Upgrade Arches 7.5 to use Python 3.11

### ISSUE THREAD

**Issue by chiatt** [archesproject/arches#9832]
Currently Arches 7.5 runs on Python 3.10, however bug fixes are no longer released to that version. 
(https://devguide.python.org/versions/)

The next Arches LTS (7.5) should use the corresponding Python stable release (3.11).

At least one change that will need to be made is the removal of 'rU' when reading files:
https://github.com/python/cpython/issues/81511

**Comment by jacobtylerwalls** [comment]
The dev dependencies use `django_nose`, but `nose` itself is no longer maintained, and will not run on Python 3.10:

```py
  File "/Users/jwalls/arches/env310/lib/python3.10/site-packages/nose/suite.py", line 106, in _set_tests
    if isinstance(tests, collections.Callable) and not is_suite:
AttributeError: module 'collections' has no attribute 'Callable'
```

Should migrating from nose to pytest be done first? At least one [automated migration tool](https://github.com/pytest-dev/nose2pytest) exists.

**Comment by jacobtylerwalls** [comment]
Ah, the nose piece is discussed in #9260 already.

**Comment by jacobtylerwalls** [comment]
@chiatt If 7.6 is now scheduled to be the LTS, should we just add support for python 3.11 in 7.5 without requiring it, and then require it in 7.6? (We could set python 3.9 as the floor for 7.5)

**Comment by chiatt** [comment]
> @chiatt If 7.6 is now scheduled to be the LTS, should we just add support for python 3.11 in 7.5 without requiring it, and then require it in 7.6? (We could set python 3.9 as the floor for 7.5)

Yeah - that sounds like a good plan. It looks like Ubuntu 23.04 ships with Python 3.11, so that should fit nicely with Arches 7.6

**Comment by jacobtylerwalls** [comment]
Fixed in #10200
